src/integrationTest/java/integration/tests/StatementTest.java

```diff
@@ -232,4 +237,92 @@ void specialIntValue() throws SQLException {
 		}
 	}
 
+	/**
+	 * This test validates that null values are sorted last.
+	 * @throws SQLException if something is going wrong
+	 * see com.firebolt.jdbc.metadata.FireboltDatabaseMetadataTest#nullSorting
+	 */
+	@Test
+	void nullSortOrder() throws SQLException {
+		try (Connection connection = createConnection();
+			 Statement statement = connection.createStatement();
+			 ResultSet rs = statement.executeQuery("select x from (select null as x union all select '' as x union all select 'a' as x) order by x")) {
+			List<String> actuals = new ArrayList<>();
+			while(rs.next()) {
```

Comment:
nit: some indent issues here and in the following functions

Reply:
I do not see any problem here right now. Let's take a look together?

Reply:
![image](https://github.com/firebolt-db/jdbc/assets/93913847/9cacac84-01ba-4cf8-be87-f86cd877c4ba)
hope this helps